src/app/widget-display/widget-display.component.ts

```diff
@@ -0,0 +1,87 @@
+import { Component, OnDestroy } from '@angular/core';
+import { FormControl } from '@angular/forms';
+import { Observable, Subscription } from 'rxjs';
+import { map, startWith } from 'rxjs/operators';
+import { WidgetComponent } from '../widget/widget.component';
+import { IWidget } from '../weather.model';
+import { WidgetApiService } from '../widget-api.service';
+import { CommonModule } from '@angular/common';
+import { MatAutocompleteModule } from '@angular/material/autocomplete';
+import { MatInputModule } from '@angular/material/input';
+import { MatFormFieldModule } from '@angular/material/form-field';
+import { MatIconModule } from '@angular/material/icon';
+import { ReactiveFormsModule } from '@angular/forms';
+
+@Component({
+  selector: 'app-widget-display',
+  standalone: true,
+  imports: [WidgetComponent, CommonModule, MatAutocompleteModule, MatInputModule, MatFormFieldModule, MatIconModule, ReactiveFormsModule],
+  templateUrl: './widget-display.component.html',
+  styleUrl: './widget-display.component.css'
+})
+export class WidgetDisplayComponent implements OnDestroy {
+  myControl = new FormControl();
+  widgets: IWidget[] = [];
+  filteredOptions: Observable<string[]>;
+  cities: string[] = [];
+
+  currentIndex: number = 0;
+  showButtons: boolean = false;
+
+  weatherSubscription: Subscription | undefined;
+
+  constructor(private widgetApiService: WidgetApiService) {
+    this.filteredOptions = this.myControl.valueChanges
+      .pipe(
+        startWith(''),
+        map(value => this._filter(value))
+      );
+
+    this.updateButtonVisibility();
+  }
+
+  _filter(value: string): string[] {
```

Comment:
symbol `_` suse for showing that it's a private method. Here it's unnecessary because it's public.